internal/server/data/migrations_test.go

```diff
@@ -850,6 +850,15 @@ INSERT INTO providers(id, name) VALUES (12345, 'okta');
 				assert.NilError(t, err)
 			},
 		},
+		{
+			label: testCaseLine("2022-11-15T10:00"),
+			expected: func(t *testing.T, tx WriteTxn) {
+				_, err := tx.Exec(`INSERT INTO access_keys(id, issued_for, name) VALUES(10000, 12345, 'foo')`)
+				assert.NilError(t, err)
+				_, err = tx.Exec(`INSERT INTO access_keys(id, issued_for, name) VALUES(10001, 12346, 'foo')`)
+				assert.NilError(t, err)
+			},
```

Comment:
Nice test coverage!

I'd expect this test to fail because it's leaving data in the tables after the test ends. I'm not sure why that safeguard is not failing the test, I'll have to look into that.

For this PR, I think this needs a `cleanup:` function set here, to delete these two rows.

Reply:
Ah, I figured out why this wasn't failing. The transaction we give to `testCase.expected` is always rolled back, so we don't technically need the `cleanup` function in this case (but it doesn't hurt to show the pattern).

Normally we would do inserts in `setup`, but this is a different case. I forgot that the transaction given to `expected` was never committed.